### PR TITLE
Updating to accept v2 values for options that have a default state. 

### PR DESCRIPTION
**Pull request by VicV**
Currently, Segment does not support sending down a 'null' value for options that have a 'default' setting (or rather, it can either be true or false or the system will decide it).

I've changed it around to accept 0,1,2 as the values, which map to an enumerator containing On/Off/Default. The taplytics SDK will handle this accordingly. 

Changed setting names to expect _V2 for these.

### CODE REVIEW

**Comment by VicV** [comment]
@f2prateek timeline on this one?

**Comment by f2prateek** [comment]
Ah sorry I didn't see this until now. I'll take a look today.

File: Segment-Taplytics/Classes/SEGTaplyticsIntegration.m [hunk 1]
```diff
@@ -168,11 +173,17 @@ - (NSNumber *)taplyticsOptionSessionBackgroundTime
     return (NSNumber *)[self.settings objectForKey:@"sessionMinutes"];
 }
 
-- (NSNumber *)shakeMenu
+- (NSNumber *)liveUpdateV2
+{
+    return (NSNumber *)[self.settings objectForKey:@"liveUpdate_v2"];
+}
+
+- (NSNumber *)shakeMenuV2
```
Comment (f2prateek) [inline Segment-Taplytics/Classes/SEGTaplyticsIntegration.m]: We're sending a string from our servers not a number.

**Review by f2prateek — CHANGES_REQUESTED** [review]

**Comment by VicV** [comment]
Ah, my bad! I'll switch it up tomorrow

**Comment by f2prateek** [comment]
Cool, and just to confirm, the strings are "true", "false" or "default".

**Comment by VicV** [comment]
@f2prateek sorry for the delay, should be good to go. Take a look.

**Comment by VicV** [comment]
@f2prateek Any chance you could take a look at this today? Need to know the timeline for a client.

File: Segment-Taplytics/Classes/SEGTaplyticsIntegration.m [hunk 2]
```diff
@@ -20,11 +20,12 @@ - (instancetype)initWithSettings:(NSDictionary *)settings
...
         [options setValue:[self delayLoad] forKey:@"delayLoad"];
-        [options setValue:[self shakeMenu] forKey:@"shakeMenu"];
         [options setValue:[self pushSandbox] forKey:@"pushSandbox"];
         [options setValue:[self taplyticsOptionSessionBackgroundTime] forKey:@"TaplyticsOptionSessionBackgroundTime"];
+        [self putDefaultBooleansWithSettings:settings withSettingsKey:@"liveUpdate_V2" andOptions:options withOptionsKey:@"liveUpdate"];
+        [self putDefaultBooleansWithSettings:settings withSettingsKey:@"shakeMenu_V2" andOptions:options withOptionsKey:@"shakeMenu"];
         
-        [self.taplyticsClass startTaplyticsAPIKey:apiKey options:settings];
+        [self.taplyticsClass startTaplyticsAPIKey:apiKey options:options];
         SEGLog(@"[[Taplytics startTaplyticsAPIKey:%@ options:%@]]", apiKey, settings);
```
Comment (f2prateek) [inline Segment-Taplytics/Classes/SEGTaplyticsIntegration.m]: This should be changed to `options` as well.

**Comment by f2prateek** [comment]
@VicV CI is failing, needs to be fixed before we can merge.

**Comment by VicV** [comment]
@f2prateek not 100% sure what the checks were failing on. Ran a pod update and sent it up. How long do these take to run?

**Comment by f2prateek** [comment]
These took about 30seconds from what I can tell https://travis-ci.org/segment-integrations/analytics-ios-integration-taplytics/builds/168721285

https://cloudup.com/cgjO99d2vQQ

**Comment by VicV** [comment]
@f2prateek I don't know what any of those failures are given that I didn't write this integration.

**Comment by f2prateek** [comment]
From what I can tell, something in your changes is breaking the CI. Are you able to build locally?

**Comment by VicV** [comment]
@f2prateek yep, no issues.

**Comment by f2prateek** [comment]
Ok i'll run it locally and see what's up.

btw this is missing new tests.

**Comment by f2prateek** [comment]
(I'll add the tests)

**Comment by f2prateek** [comment]
Closed by #8

**Comment by f2prateek** [comment]
1.1.0 with these changes is out now.

**Pull request closed without merging**